### DISCUSSION TITLE
trpc with splitAtom

### DISCUSSION THREAD

**Discussion by yudyananda** [pmndrs/jotai#1490] [Q&A] [answered]
i'm using `trpc` and [create-t3-app](https://github.com/t3-oss/create-t3-app). I wonder how to create `splitAtom` from the `trpc` query  with `jotai-trpc`, Have no idea how to make it work.

```js
import { createAtomCreators } from 'jotai-trpc';

const {  } = createAtomCreators<AppRouter>({

});

const data = 

```

**Comment by dai-shi** [comment, marked as answer]
What's your problem? Are you already familiar with splitAtom?

```ts
import { createAtomCreators } from 'jotai-trpc';

const { atomWithQuery } = createAtomCreators<AppRouter>(...);

const dataAtom = atomWithQuery(...);

const itemsAtom = splitAtom(dataAtom);
```

**Reply by yudyananda** [reply]
Frankly speaking i'm new with jotai and trpc but i already read about [splitAtom](https://jotai.org/docs/advanced-recipes/large-objects) and made same experiment with the `initialData` you provide on the doc. 

The problem comes when trying to replace initialData with data retrieved using trpc. I thought it would be as simple as `const data = atom(trpc.someRoute.someData.useQuery().data)`. Luckily I found `jotai-trpc` but don't  how to use it yet.

try to fill the `transformer` and `links` with the same value from my default trpc client but stuck in `atomWithQuery` which i'm not sure what kind of data i have to pass in it.

```js

const { atomWithQuery } = createAtomCreators<AppRouter>({
  transformer: superjson,
  links: [
    loggerLink({
      enabled: (opts) =>
        process.env.NODE_ENV === 'development' ||
        (opts.direction === 'down' && opts.result instanceof Error),
    }),
    httpBatchLink({
      url: `${baseUrl}/api/trpc`,
    }),
  ],
});

const data = atomWithQuery(?)

```

**Reply by dai-shi** [reply]
Sounds like your question is more on jotai-trpc usage.
Does this example help?
https://github.com/jotai-labs/jotai-trpc/blob/f56509e971db886c8c12f906e00ba792deef7879/examples/01_typescript/src/App.tsx#L25

**Reply by yudyananda** [reply]
Ok so i have to pass the path right? somehow intelisense does not give me much option with the string. I wonder if this related with tRPC version? i;m using v.10.

anyway you can check the repo **[here](https://github.com/yudyananda/jota-trpc-experiment)**

**Reply by dai-shi** [reply]
Ah, it's still v9. I'm yet to work on v10. https://github.com/jotai-labs/jotai-trpc/pull/6

**Reply by yudyananda** [reply]
that makes sense, thanks for your time
